nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -320,22 +324,69 @@ def __init__(
             self.pad_value = params['pad_value'] if 'pad_value' in params else 0.0
 
         with self.pipe:
-            audio, indices = dali.fn.readers.nemo_asr(
-                name="Reader",
-                manifest_filepaths=manifest_filepath.split(','),
-                dtype=dali.types.FLOAT,
-                downmix=True,
-                sample_rate=float(self.sample_rate),
-                min_duration=min_duration,
-                max_duration=max_duration,
-                read_sample_rate=False,
-                read_text=False,
-                read_idxs=True,
-                random_shuffle=shuffle,
-                shard_id=self.shard_id,
-                num_shards=self.num_shards,
-                pad_last_batch=True,
-            )
+            if audio_tar_filepaths is None and audio_tar_index_filepaths is None:
+                audio, indices = dali.fn.readers.nemo_asr(
+                    name="Reader",
+                    manifest_filepaths=manifest_filepath.split(','),
+                    dtype=dali.types.FLOAT,
+                    downmix=True,
+                    sample_rate=float(self.sample_rate),
+                    min_duration=min_duration,
+                    max_duration=max_duration,
+                    read_sample_rate=False,
+                    read_text=False,
+                    read_idxs=True,
+                    random_shuffle=shuffle,
+                    shard_id=self.shard_id,
+                    num_shards=self.num_shards,
+                    pad_last_batch=True,
+                )
+
+                self.is_tarred_dataset = False
+
+            elif audio_tar_filepaths is not None and audio_tar_index_filepaths is not None:
```

Comment:
I would make `audio_tar_index_filepaths` optional as DALI can build index on the fly.
You can add a check if it is empty or length is equal to audio_tar_filepaths.

Reply:
Hmm I actually want to avoid the scenario where Dali silently builds index. Since tarred dataset is for large datasets, and index building on millions of files takes non trivial time, I'd rather enforce the user to supply preprocessed indices.

Reply:
Sure. DALI shows a warning if you don't provide the index but it is not a technical blocker.
Up to you.